File: Backend/todo/src/main/java/com/proj/todo/service/TaskService.java
package com.proj.todo.service;


import com.proj.todo.model.Task;
import com.proj.todo.repository.TaskRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class TaskService {
    @Autowired
    private TaskRepository taskRepository;

    public List<Task> getAllTasks() {
        return taskRepository.findAll();
    }

    public Task createTask(Task task) {
        return taskRepository.save(task);
    }

    public Task updateTask(Long id, Task updatedTask) {
        Task existingTask = taskRepository.findById(id).orElseThrow(() -> new RuntimeException("Task not found"));
        //existingTask.setName(updatedTask.getName());
        existingTask.setDone(updatedTask.getDone());
        return taskRepository.save(existingTask);
    }

    public List<Task> getTasksByDone(boolean done) {
        return taskRepository.findByDone(done);
    }

    public List<Task> getTasksSortedByDate(String order) {
        if ("asc".equalsIgnoreCase(order)) {
            return taskRepository.findByOrderByCreatedAtAsc();
        } else {
            return taskRepository.findByOrderByCreatedAtDesc();
        }
    }
}

File: src/components/TaskService.js
import api from "../services/api";

export const getTasks = async () => {
  return await api.get("/tasks");
};

export const createTask = async (task) => {
  return await api.post("/tasks", task);
};

export const updateTask = async (id, updatedTask) => {
  return await api.put(`/tasks/${id}`, updatedTask);
};

export const getTasksByDone = async (done) => {
  return await api.get(`/tasks/done?done=${done}`);
};

export const getTasksSortedByDate = async (order) => {
  return await api.get(`/tasks/sorted?order=${order}`);
};
